qt/widgets/plotting/src/Qwt/PreviewPlot.cpp

```diff
@@ -987,3 +995,20 @@ void PreviewPlot::handleAxisTypeSelect() {
   // Update the plot
   emit needToHardReplot();
 }
+
+/**
+ *
+ * Removes the Y-Axis option from the context menu as the Sample Transmission
+ * Calculator does not need it.
+ *
+ * @throw Runtime error thrown if the Y Axis action can't be found by its object
+ * name
+ */
+void PreviewPlot::disableYAxisMenu() {
+  auto menu = m_contextMenu->findChild<QAction *>(g_yAxisMenuName);
+
+  if (menu == NULL)
+    throw std::runtime_error("There was a problem with object naming.");
```

Comment:
Can this be made more specific, something like "Y-Axis menu object cannot be retrieved"?